bus-mapping/src/circuit_input_builder.rs

```diff
@@ -888,11 +891,25 @@ impl<'a> CircuitInputStateRef<'a> {
 
     /// Push a new [`Call`] into the [`Transaction`], and add its index and
     /// [`CallContext`] in the `call_stack` of the [`TransactionContext`]
-    pub fn push_call(&mut self, call: Call) {
-        let call_id = call.call_id;
+    pub fn push_call(&mut self, call: Call, step: &GethExecStep) {
+        let call_data_source = match call.kind {
```

Comment:
I thought about that, but while generating the memory read, the `memory_address` has not offset, so I think it's more straightforward to just have the complete caller memory, to avoid carrying the `call_data_offset` and adding it for every read.

Reply:
I see, thanks for the explanation!  Sounds good then to use `call_data_source`